src/ai/backend/agent/stats.py

```diff
@@ -454,3 +475,105 @@ async def _pipe_builder(r: Redis) -> Pipeline:
             return pipe
 
         await redis_helper.execute(self.agent.redis_stat_pool, _pipe_builder)
+
+    async def collect_per_container_process_stat(
+        self,
+        container_ids: Sequence[ContainerId],
+    ) -> None:
+        """
+        Collect the per-container process statistics only,
+
+        Intended to be used by the agent.
+        """
+        async with self._lock:
+            pid_map = {}
+            total_pids = []
+            pids = []
+            for cid in container_ids:
+                try:
+                    docker = aiodocker.Docker()
+                    result = await docker._query_json(f"containers/{cid}/top", method="GET")
+                    procs = result["Processes"]
+                    pids = [PID(int(proc[1])) for proc in procs if proc[0] != "root"]
+                    if os.getuid() == 0:
+                        pids.extend([PID(int(proc[1])) for proc in procs if proc[0] == "root"])
+                    unused_pids = set(self.process_metrics[cid].keys()) - set(pids)
+                except (KeyError, aiodocker.exceptions.DockerError):
+                    log.warning(
+                        "collect_per_container_process_stat(): cannot found container {}", cid
+                    )
```

Comment:
Not finding container with preserved ID is nothing to warn about, since the container always has chance to gracefully exit when user terminates kernel. Let's change log level to "debug" instead of "warn" not to spam user with the logging.